db/db_tunables.c

```diff
@@ -869,6 +871,19 @@ static int page_order_table_scan_update(void *context, void *value)
     return 0;
 }
 
+static void* file_permissions_value(void *context) {
+    static char val[15];
+    snprintf(val, sizeof(val), "0%o", gbl_file_permissions);
+    return val;
+}
+
+static int file_permissions_update(void *context, void *value) {
+    (void)context;
+    char *in = (char*) value;
+    gbl_file_permissions = strtol(in, NULL, 8);
```

Comment:
I think we should validate (errno, valid permission) the input before updating the global variable. Also log a line about this change.